services/payments/src/simcore_service_payments/db/payment_users_repo.py

```diff
@@ -10,13 +13,53 @@ class PaymentsUsersRepo(BaseRepository):
     # Currently linked to `users` but expected to be linked to `payments_users`
     # when databases are separated. The latter will be a subset copy of the former.
     #
-    async def get_primary_group_id(self, user_id: UserID) -> GroupID:
+
+    async def _get(self, query):
         async with self.db_engine.begin() as conn:
-            result = await conn.execute(
-                sa.select(users.c.primary_gid).where(users.c.id == user_id)
-            )
-            row = result.first()
-            if row is None:
-                msg = f"{user_id=} not found"
-                raise ValueError(msg)
+            result = await conn.execute(query)
+            return result.first()
+
+    async def get_primary_group_id(self, user_id: UserID) -> GroupID:
+        if row := await self._get(
+            sa.select(
+                users.c.primary_gid,
+            ).where(users.c.id == user_id)
+        ):
             return GroupID(row.primary_gid)
+
+        msg = f"{user_id=} not found"
+        raise ValueError(msg)
+
+    async def get_notification_data(self, user_id: UserID, payment_id: PaymentID):
+        """Retrives data that will be injected in a notification for the user on this payment"""
```

Comment:
You know our discussion about one or two doors decisions :) This one is for me a bit "one door" as you are breaking the initial thought to have payments and resource usage data decoupled from other tables as we can then easily migrate to second database. Or when we introduce separate notification service, this will be moved there and will not be needed here anymore?

Reply:
:-) yes, I aware of that the whole `payment_users_repo` violates that.  And yes, I have put it all in one place because the idea here is to move the part of the "user+product" this to the notification service. 

@matusdrobuliak66 I could also extend the payments db with the info on users and product missing and send this information from the webserver on every payment. Can you think of any other way?

Reply:
ok thanks! not now, but we can brainstorm sometimes together.